_schemas/scripts/validator.js

```diff
@@ -0,0 +1,204 @@
+// Import the modules
+const Ajv = require("ajv");
+const Ajv2020 = require("ajv/dist/2020");
+const addFormats = require("ajv-formats");
+const readdirp = require("readdirp");
+const npath = require("path");
+const compareVersions = require("compare-versions");
+
+// Define root dir
+const rootDir = "./_schemas";
+
+// Define object that will record the full path of all files tested
+const files = {};
+
+// Check for arguments
+const verbose = process.argv.indexOf("-v") > -1;
+
+// Main function
+(async function () {
+  try {
+    // List all the available versions of the spec
+    for await (const entry of readdirp(rootDir, { depth: 0, entryType: "directories" })) {
+      const version = entry.path;
+      if (isNaN(version)) {
+        continue;
+      }
+
+      // For this version, ...
+      console.log("Version : %s", version);
+
+      // ... create ajv instance
+      const ajv = ajvFactory(version);
+
+      // ... add schemas
+      await addSchemas(ajv, version);
+
+      // ... prepare record object
+      files[version] = new Array();
+
+      // ... test all files
+      const { ok, ko } = await handleTests(ajv, version);
+
+      // ... log the results
+      if (ko.length != 0) {
+        console.log("Errors : %d / %d", ko.length, ko.length + ok.length);
```

Comment:
Currently the output in case of errors is very limited:

```
$ node ./_schemas/scripts/validator.js
Version : 1.0
Errors : 1 / 94
```

It would be great if it would log more information about the error. Especially, which test is failing and why. Maybe something like this:

```
Version : 1.0

request/resource/create/valid/post_resource.json should be valid but is invalid.
  /data/id must be string

Errors : 1 / 94
```

The needed data seems to be already available in `ko` objects:

```json
{
  version: '1.0',
  relativePath: 'request/resource/create/valid/post_resource.json',
  fullPath: '/home/main/tmp/json-api/_schemas/1.0/tests/request/resource/create/valid/post_resource.json',
  ok: false,
  errors: [
    {
      instancePath: '/data/id',
      schemaPath: 'https://jsonapi.org/schemas/spec/v1.0/draft#/definitions/resourceIdentificationNew/properties/id/type',
      keyword: 'type',
      params: [Object],
      message: 'must be string'
    }
  ]
}
```

Maybe the output of `test` command from `ajv-cli` could give some inspiration: https://ajv.js.org/packages/ajv-cli.html#test-validation-result

Reply:
You're right.
I improved script outputs.